### ISSUE TITLE
Can't start server: unexpected keyword argument 'address'

### ISSUE THREAD

**Issue by screendriver** [nicolargo/glances#1693]
## Description

I installed `glances` on my new Raspberry Pi 4 via

```sh
$ pip3 install 'glances[action,browser,cloud,cpuinfo,docker,export,folders,gpu,graph,ip,raid,snmp,web,wifi]'
```

`glances` itself works as expected but when I launch it in server mode I get following error

```sh
➜  ~ glances -s
Glances XML-RPC server is running on 0.0.0.0:61209
Traceback (most recent call last):
  File "/home/pi/.local/bin/glances", line 10, in <module>
    sys.exit(main())
  File "/home/pi/.local/lib/python3.7/site-packages/glances/__init__.py", line 143, in main
    start(config=config, args=args)
  File "/home/pi/.local/lib/python3.7/site-packages/glances/__init__.py", line 108, in start
    mode = GlancesMode(config=config, args=args)
  File "/home/pi/.local/lib/python3.7/site-packages/glances/server.py", line 229, in __init__
    self.autodiscover_client = GlancesAutoDiscoverClient(socket.gethostname().split('.', 1)[0], args)
  File "/home/pi/.local/lib/python3.7/site-packages/glances/autodiscover.py", line 215, in __init__
    port=args.port, weight=0, priority=0, properties={}, server=hostname)
TypeError: __init__() got an unexpected keyword argument 'address'
```

But when I disable autodiscover it works: `glances -s --disable-autodiscover`

#### Versions

* Glances & psutil (glances -V): v3.1.4.1 with PsUtil v5.5.1
* Operating System (lsb_release -a):

Distributor ID:	Raspbian
Description:	Raspbian GNU/Linux 10 (buster)
Release:	10
Codename:	buster

**Comment by nicolargo** [comment]
Hi @screendriver 

What is your zeroconf version ?

```
$ python
>>> import zeroconf
>>> zeroconf.__version__
0.24.4
```

**Comment by screendriver** [comment]
`0.28.0`

**Comment by LinuxOnTheDesktop** [comment]
I have a similar problem on Mint 20 using Glances v3.1.3 with psutil v5.5.1. I installed Glances from the distro's repository. The error message (or part of it) is:

`TypeError: __init__() got an unexpected keyword argument 'address'`

Possibly relevant might be this (which I did to try to get the 'zeroconf' version):

```
$ python
bash: python: command not found
$ which python3
/usr/bin/python3
$ which python2
$
```

**Comment by velmirslac** [comment]
I'm getting the same error on RHEL 7.8

```
$ glances -s
Glances XML-RPC server is running on 0.0.0.0:61209
Traceback (most recent call last):
  File "/usr/local/bin/glances", line 11, in <module>
    load_entry_point('Glances==3.1.4.1', 'console_scripts', 'glances')()
  File "/usr/local/lib/python3.6/site-packages/glances/__init__.py", line 143, in main
    start(config=config, args=args)
  File "/usr/local/lib/python3.6/site-packages/glances/__init__.py", line 108, in start
    mode = GlancesMode(config=config, args=args)
  File "/usr/local/lib/python3.6/site-packages/glances/server.py", line 229, in __init__
    self.autodiscover_client = GlancesAutoDiscoverClient(socket.gethostname().split('.', 1)[0], args)
  File "/usr/local/lib/python3.6/site-packages/glances/autodiscover.py", line 215, in __init__
    port=args.port, weight=0, priority=0, properties={}, server=hostname)
TypeError: __init__() got an unexpected keyword argument 'address'

$ glances --version
Glances v3.1.4.1 with PsUtil v5.7.2
Log file: /home/velmir/.local/share/glances/glances.log

$ python3
Python 3.6.8 (default, Sep 26 2019, 11:57:09) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-39)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import zeroconf
>>> zeroconf.__version__
'0.28.0'
```

**Comment by nicolargo** [comment]
Following backwards incompatibilities should be taken into account:
- ifaddr 0.1.7 or newer is required now (0.28.0)
- Removed previously deprecated ServiceInfo address constructor parameter and property (0.27.0)
- Rationalized handling of TXT records. Non-bytes values are converted to str and encoded to bytes using UTF-8 now, None values mean value-less attributes. When receiving TXT records no decoding is performed now, keys are always bytes and values are either bytes or None in value-less attributes. (0.25.0)

**Comment by nicolargo** [comment]
Corrected in the previous commit (67bf668) on the DEVELOP branch. 

Glances is now compatible with all zeroconf version.

**Comment by sebma** [comment]
Hi everyone, I had the same issue on Ubuntu 20.04 LTS and here is what I typed to fixed my `zeroconf` configuration : 

```bash
$ python3 -c "import zeroconf;print(zeroconf.__version__)"
0.38.7
$ python3 -c "import zeroconf;print(zeroconf.__path__)"
['/usr/local/lib/python3.8/dist-packages/zeroconf']
$ glances -V
Glances v3.1.3 with psutil v5.5.1
$ sudo -H pip3 uninstall zeroconf
Found existing installation: zeroconf 0.38.7
Uninstalling zeroconf-0.38.7:
  Would remove:
    /usr/local/lib/python3.8/dist-packages/zeroconf-0.38.7.dist-info/*
    /usr/local/lib/python3.8/dist-packages/zeroconf/*
Proceed (Y/n)? y
  Successfully uninstalled zeroconf-0.38.7
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
$ sudo apt install -V python3-zeroconf
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
   python3-ifaddr (0.1.6-1)
   python3-nose (1.3.7-5)
Suggested packages:
   python-nose-doc (1.3.7-5)
The following NEW packages will be installed:
   python3-ifaddr (0.1.6-1)
   python3-nose (1.3.7-5)
   python3-zeroconf (0.24.4-0ubuntu1)
0 upgraded, 3 newly installed, 0 to remove and 2 not upgraded.
Need to get 164 kB of archives.
After this operation, 786 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://fr.archive.ubuntu.com/ubuntu focal/universe amd64 python3-ifaddr all 0.1.6-1 [7 492 B]
Get:2 http://fr.archive.ubuntu.com/ubuntu focal/universe amd64 python3-nose all 1.3.7-5 [116 kB]
Get:3 http://fr.archive.ubuntu.com/ubuntu focal/universe amd64 python3-zeroconf all 0.24.4-0ubuntu1 [40,6 kB]
Fetched 164 kB in 0s (776 kB/s)              
Selecting previously unselected package python3-ifaddr.
(Reading database ... 377923 files and directories currently installed.)
Preparing to unpack .../python3-ifaddr_0.1.6-1_all.deb ...
Unpacking python3-ifaddr (0.1.6-1) ...
Selecting previously unselected package python3-nose.
Preparing to unpack .../python3-nose_1.3.7-5_all.deb ...
Unpacking python3-nose (1.3.7-5) ...
Selecting previously unselected package python3-zeroconf.
Preparing to unpack .../python3-zeroconf_0.24.4-0ubuntu1_all.deb ...
Unpacking python3-zeroconf (0.24.4-0ubuntu1) ...
Setting up python3-nose (1.3.7-5) ...
Setting up python3-ifaddr (0.1.6-1) ...
Setting up python3-zeroconf (0.24.4-0ubuntu1) ...
Processing triggers for man-db (2.9.1-1) ...
$ glances -s
Glances XML-RPC server is running on 0.0.0.0:61209
Announce the Glances server on the LAN (using 192.168.1.6 IP address)
^C
$ sudo systemctl restart glances.service
$ systemctl status glances.service
● glances.service - Glances
     Loaded: loaded (/lib/systemd/system/glances.service; enabled; vendor preset: enabled)
     Active: active (running) since Sun 2022-08-21 15:05:09 CEST; 3min 43s ago
       Docs: man:glances(1)
             https://github.com/nicolargo/glances
   Main PID: 7309 (glances)
      Tasks: 3 (limit: 4556)
     Memory: 39.0M
     CGroup: /system.slice/glances.service
             └─7309 /usr/bin/python3 /usr/bin/glances -s -B 127.0.0.1

août 21 15:05:09 myPC systemd[1]: Started Glances.
$
```
